File: python/tower_of_hanoi/TowerOfHanoi.py
#!/usr/bin/python3

def tower_of_hanoi(n:int, fromPeg:str, toPeg:str, auxPeg:str):
    if n==1:
        print(f"Move disk from {fromPeg} to {toPeg}")
        return
    tower_of_hanoi(n-1, fromPeg, auxPeg, toPeg)
    print(f"Move disk from {fromPeg} to {toPeg}")
    tower_of_hanoi(n-1, auxPeg, toPeg, fromPeg)

print("Executing for 1 disk")
tower_of_hanoi(1,"A","B","C")
print("Executing for 2 disks")
tower_of_hanoi(2,"A","B","C")
print("Executing for 3 disks")
tower_of_hanoi(3,"A","B","C")

File: java/tower_of_hanoi/TowerOfHanoi.java
class TowerOfHanoi{
    public void calculate(int n, char fromPeg, char toPeg, char auxPeg){
        if(n==1){
            System.out.println("Move disk from "+ fromPeg + " to " + toPeg);
            return;
        }
        calculate(n-1, fromPeg, auxPeg, toPeg);
        System.out.println("Move disk from "+ fromPeg + " to " + toPeg);
        calculate(n-1, auxPeg, toPeg, fromPeg);
    }
    public static void main(String args[]){
        TowerOfHanoi t = new TowerOfHanoi();
        System.out.println("Executing for 1 disk");
        t.calculate(1,'A','B','C');
        System.out.println("Executing for 2 disks");
        t.calculate(2,'A','B','C');
        System.out.println("Executing for 3 disks");
        t.calculate(3,'A','B','C');
    }
}